src/View/Helper/FormHelper.php

```diff
@@ -1445,6 +1447,15 @@ protected function _magicOptions($fieldName, $options, $allowOverride)
             $options['required'] = $context->isRequired($fieldName);
         }
 
+        if (method_exists($context, 'getRequiredMessage')) {
+            $message = $context->getRequiredMessage($fieldName);
+            if ($options['required'] && $message && $this->getConfig('useValidationMessages')) {
+                $message = h(addslashes($message));
+                $options['oninvalid'] = "this.setCustomValidity('$message')";
```

Comment:
No, but I can make one. Another option would be to always include the message in a `data-` attribute, that way users can disregard or use it.

Reply:
Use `data-` attributes sounds good too?

Reply:
What about a string based template much like the form helper uses otherwise (that default template could then use data or not, I don't personally care much, as it is configurable then)?

Reply:
@inoas that's not a bad idea. I would have to place it into the control's `templateVars` so users could add `'input' => '<input type="{{type}}" name="{{name}}"{{attrs}} data-mysuperawesomeattribute="{{requiredMessage}}"/>',`.

Reply:
@ADmad I included `requiredMessage` in the templateVars if the field is "required" and has a message. This allows you to adjust the templates according to whatever data attribute you'd want, per @ionas's suggestion.

Part of the documentation I write after this is merged will include this change, and an example on using it.